components/neeva/browser/content_filter_stats.cc

```diff
@@ -2,13 +2,51 @@
 
 #include "components/neeva/browser/content_filter_stats.h"
 
+#include "components/neeva/browser/content_filter_rules_config.h"
+
 namespace neeva {
 
 ContentFilterStats::~ContentFilterStats() = default;
 
-void ContentFilterStats::RecordFilteredHost(const std::string& host) {
+void ContentFilterStats::RecordFilteredHost(
+    const std::string& rules_file, const std::string& host) {
   // TODO: Consider mapping a.foo.com to foo.com here instead of downstream.
-  hosts_to_counts_[host] += 1;
+  data_[rules_file][host] += 1;
+}
+
+void ContentFilterStats::GetHostsForFilter(
+    const std::string& rules_file, std::vector<std::string>* hosts) const {
+  if (rules_file != "*") {
+    auto it = data_.find(rules_file);
+    if (it != data_.end()) {
+      for (const auto& entry : it->second) {
+        hosts->push_back(entry.first);
+      }
+    }
+  } else {
+    for (const auto* filename : ContentFilterRulesConfig::kValidRulesFiles) {
+      GetHostsForFilter(filename, hosts);
+    }
+  }
+}
+
+int ContentFilterStats::GetHostCountsForFilter(
+    const std::string& rules_file, const std::string& host) const {
+  int count = 0;
+  if (rules_file != "*") {
+    auto it = data_.find(rules_file);
+    if (it != data_.end()) {
+      auto result = it->second.find(host);
+      if (result != it->second.end()) {
+        count = result->second;
+      }
```

Comment:
That's a neat idea. I'll explore.

Reply:
I looked into it. I think it will result in code that is harder to follow to be honest. If we had a lot of code following this pattern, then it would make more sense. But for just these two functions, it seems like the complexity isn't worth it.

Reply:
Sounds good